tests/core/test_scripts.py

```diff
@@ -3092,6 +3098,68 @@ def test_do_run(self):
         assert 1 == customlist.size
 
 
+class TestCustomListUpdateEntriesScript(EndToEndSearchTest):
+    def populate_works(self):
+        self.populated_books: list[Work] = [
+            self._work(with_license_pool=True, title="Populated Book") for _ in range(5)
+        ]
+        self.unpopular_books: list[Work] = [
+            self._work(with_license_pool=True, title="Unpopular Book") for _ in range(3)
+        ]
+
+    def test_process_custom_list(self):
+        last_updated = datetime.datetime.now() - datetime.timedelta(hours=1)
+        custom_list, _ = self._customlist()
+        custom_list.library = self._default_library
+        custom_list.auto_update_enabled = True
+        custom_list.auto_update_query = json.dumps(
+            dict(query=dict(key="title", value="Populated Book"))
+        )
+        custom_list.auto_update_last_update = last_updated
+
+        custom_list1, _ = self._customlist()
+        custom_list1.library = self._default_library
+        custom_list1.auto_update_enabled = True
+        custom_list1.auto_update_query = json.dumps(
+            dict(query=dict(key="title", value="Unpopular Book"))
+        )
+        custom_list1.auto_update_last_update = last_updated
+
+        # Do the process
+        script = CustomListUpdateEntriesScript(self._db)
+        mock_parse = MagicMock()
+        mock_parse.return_value.libraries = [self._default_library]
+        script.parse_command_line = mock_parse
+
+        with freeze_time("2022-01-01") as frozen_time:
+            script.run()
+
+        self._db.refresh(custom_list)
+        self._db.refresh(custom_list1)
+        assert len(custom_list.entries) == 1 + len(
+            self.populated_books
```

Comment:
This is not really related to this PR but why is `len(custom_list.entries) = 1 + ..`? Does this mean the empty list has one entry?

Reply:
The testing method _customlist() creates a list with one work in it already